### PR TITLE
Informative error message in development

### PR DESCRIPTION
**Pull request by varunzxzx**
It should handle the error gracefully. Getting correct error message will speed up the debugging process. Error thrown by any module will be handled gracefully by the error handler and will return the appropriate error code and the cause of the error.

### CODE REVIEW

File: stackle_api/server.js [hunk 1]
```diff
@@ -42,4 +42,18 @@ mongoose.connect(database.url,function(err){
...
+    var err = new Error("Not Found");
+    err.status = 404;
+    next(err);
+});
+  
+// error handler
+app.use(function(err, req, res, next) {
+    // render the error page
+    res.status(err.status || 500);
+    process.env.NODE_ENV !== "DEVELOPMENT" ? res.sendFile(__dirname + "/public/404.html") : res.send(`<h1>Error Code: ${err.status || 500}</h1><br><p>${err.stack}</p>`);
```
Comment (agentmilindu) [inline stackle_api/server.js]: Can we show `404` when that page or resource is not found and show a `5xx` page for all the other errors?

**Comment by agentmilindu** [comment]
@psnmissaka @tharindupr Can we merge this PR?

File: stackle_api/server.js (hunk 1 above)
Comment (varunzxzx) [reply stackle_api/server.js]: Yes, we can do that also.

**Comment by varunzxzx** [comment]
@padamchopra middleware on the line [46](https://github.com/scorelab/Stackle/pull/75/commits/8f0ba9ec0651bb5a8c409bcc844881322cde6d24#diff-c17882ac399a26e32d27745d30fc1346R46) is handling the 404 and passing the error to the error handler which is on the line [53](https://github.com/scorelab/Stackle/pull/75/commits/8f0ba9ec0651bb5a8c409bcc844881322cde6d24#diff-c17882ac399a26e32d27745d30fc1346R53)

**Comment by agentmilindu** [comment]
@varunzxzx But In PROD we still see the `404.html` even for 5xx errors, right?

**Comment by varunzxzx** [comment]
Yes, that's true.

**Comment by varunzxzx** [comment]
@agentmilindu fixed the issue.

**Comment by varunzxzx** [comment]
PR ready for review

**Comment by varunzxzx** [comment]
@iammosespaulr Fixed!

**Pull request merged**